src/shared/ofi_str.c

```diff
@@ -199,3 +200,50 @@ void ofi_free_string_array(char **s)
 	/* and then the actual array of pointers */
 	free(s);
 }
+
+char *ofi_tostr_size(char *str, size_t len, uint64_t size)
+{
+	uint64_t base = 0;
+	uint64_t fraction = 0;
+	char mag;
+
+	memset(str, '\0', len);
+	if (size >= (1 << 30)) {
+		base = 1 << 30;
+		mag = 'G';
+	} else if (size >= (1 << 20)) {
+		base = 1 << 20;
+		mag = 'M';
+	} else if (size >= (1 << 10)) {
+		base = 1 << 10;
+		mag = 'K';
+	} else {
+		base = 1;
+		mag = '\0';
+	}
+
+	if (size / base < 10)
+		fraction = (size % base) * 10 / base;
+
+	if (fraction)
+		snprintf(str, len, "%lu.%lu%c", size / base, fraction, mag);
+	else
+		snprintf(str, len, "%lu%c", size / base, mag);
```

Comment:
We may want to match the behavior of the other ofi_tostr... functions that are in fi_tostr.c.  Those call ofi_strncatf() instead of snprintf(), and concatenate the new string to the end of the current one, rather than replacing the existing string.

That may match the expected use case for these functions as well.